IPython/core/pylabtools.py

```diff
@@ -319,7 +333,23 @@ def find_gui_and_backend(gui=None, gui_select=None):
 
     import matplotlib
 
-    has_unified_qt_backend = getattr(matplotlib, "__version_info__", (0, 0)) >= (3, 5)
+    if _matplotlib_manages_backends():
+        backend_registry = matplotlib.backends.registry.backend_registry
```

Comment:
Does that need explicit import of `matplotlib.backends` ? and/or other hasattr checks ?

```
In [1]: import matplotlib

In [2]: matplotlib.backends
...
AttributeError: module 'matplotlib' has no attribute 'backends'

In [3]: import matplotlib.backends
```

Reply:
The `_matplotlib_manages_backends()` call checks that the version of Matplotlib is late enough to include the required `backed_registry` functions, so the import *should* always work. But now you have drawn my attention to this I realise it would be better to be explicit and use `hasattr` checks for `backend_registry` and its function `resolve_gui_or_backend`. This would avoid us having to have a hardcoded Matplotlib version and would allow us to merge this work before we are entirely sure what version of Matplotlib will include the corresponding changes.